src/main/java/pl/agh/edu/master_diet/core/model/shared/SearchResult.java

```diff
@@ -0,0 +1,18 @@
+package pl.agh.edu.master_diet.core.model.shared;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.List;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Setter
+@Getter
+public class SearchResult<T> {
```

Comment:
Btw, wouldn't be ```@Builder``` useful in here?

Reply:
There was a problem with the generic builder. I have no idea how to make it work with annotation.